app/src/main/java/ai/elimu/appstore/BaseApplication.java

```diff
@@ -92,13 +102,55 @@ public Retrofit getRetrofit() {
         HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
         logging.setLevel(BuildConfig.DEBUG ? HttpLoggingInterceptor.Level.BODY :
                 HttpLoggingInterceptor.Level.NONE);
-        final OkHttpClient okHttpClient = new OkHttpClient.Builder()
+
+        final ProgressResponseBody.ProgressListener progressListener = new ProgressResponseBody.ProgressListener() {
+            @Override
+            public void update(long bytesRead, long contentLength, boolean done) {
```

Comment:
How frequently is this method called? Every 1024 bytes?

Reply:
It's not 1024 bytes but a random number, which is the number of bytes it receives from server.
Kindly refer to the class here
https://github.com/square/okhttp/blob/209c6c5589d7b0a0f10b4ce78ac6be0740a5cea8/samples/guide/src/main/java/okhttp3/recipes/Progress.java#L68